corehq/apps/fixtures/fixturegenerators.py

```diff
@@ -43,37 +47,59 @@ class ItemListsProvider(FixtureProvider):
 
     def __call__(self, restore_state):
         restore_user = restore_state.restore_user
+        domain = restore_user.domain
+        db = get_blob_db()
 
-        all_types = {t._id: t for t in FixtureDataType.by_domain(restore_user.domain)}
+        all_types = {t._id: t for t in FixtureDataType.by_domain(domain)}
         global_types = {id: t for id, t in all_types.items() if t.is_global}
 
-        items_by_type = defaultdict(list)
+        if restore_state.overwrite_cache:
+            db.delete(domain, FIXTURE_BUCKET)
+
+        # get global types from the db
+        try:
+            global_items = [db.get(domain, FIXTURE_BUCKET).read()]
+        except NotFound:
+            global_items = self.get_global_items(restore_user, global_types)
+            cached_element = [ElementTree.tostring(element, encoding='utf-8') for element in global_items]
+            db.put(StringIO("".join(cached_element)), domain, FIXTURE_BUCKET)
```

Comment:
Technically, this is storing invalid XML, since it has no root node. The [restore class](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/restore.py#L143-L144) just blindly appends this into the `<OpenRosaResponse>`, so this isn't an issue, however we do have to deal with it when looking up the [fixture by id](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/fixtures.py#L87-L94) (this is only called by cloudcare in very specific instances).

Another solution I explored was saving the cached xml as:
```<cached-fixture>...<cached-fixture>```
and when fetching it from the blobdb:
```py
cached_items = ElementTree.fromstring(db.get(domain, FIXTURE_BUCKET).read())
global_items = [item for item in cached_items]
```
However, this was much slower than just storing it as string and doing no manipulation at all (0.02s vs 0.5s for the eNikshay ItemLists locally)